file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt

```diff
@@ -578,12 +574,87 @@ class AttributeCachingFileSystemTests {
+            Files.newOutputStream(cachingPath).use { outputStream ->
+                outputStream.write("hello".toByteArray(Charsets.UTF_8))
+            }
+
+            assertThat(cachingPath).isInstanceOf(AttributeCachingPath::class.java)
+            assertThat((cachingPath as AttributeCachingPath).isCachedInitialized()).isFalse()
+            // Force cachingPath initialization after the file is created with getPath on the path's string representation.
+            // This test requires that the cache of the file under test be initialized.
+            cachingPath = it.getPath(cachingPath.toString())
+            assertThat((cachingPath as AttributeCachingPath).isCachedInitialized()).isTrue()
+
+            try {
+                // Test with original file owner on default filesystem because it's a large amount of work to create our
+                // own test user there.
+
+                val originalAttributeView = Files.getFileAttributeView(cachingPath, AclFileAttributeView::class.java)
+                val originalOwner = originalAttributeView.owner
+                val originalAclEntries = originalAttributeView.acl
+
+                // simulate concurrent modification on default filesystem
+                val concurrentPath = defaultFileSystem.getPath(
+                    "$tempDirPath${defaultFileSystem.separator}testfile-$uniqueID.txt",
+                )
+
+                val acl = AclEntry.newBuilder()
+                    .setType(AclEntryType.ALLOW)
+                    .setPrincipal(originalOwner)
+                    .setFlags(AclEntryFlag.FILE_INHERIT)
+                    .setPermissions(
+                        AclEntryPermission.WRITE_NAMED_ATTRS,
+                        AclEntryPermission.WRITE_ATTRIBUTES,
+                        AclEntryPermission.WRITE_DATA,
+                        AclEntryPermission.READ_ACL,
+                        AclEntryPermission.APPEND_DATA,
+                        AclEntryPermission.READ_ATTRIBUTES,
+                        AclEntryPermission.READ_DATA,
+                        AclEntryPermission.READ_NAMED_ATTRS,
+                        AclEntryPermission.DELETE,
+                    )
+                    .build()
+                Files.setAttribute(concurrentPath, "acl:acl", listOf<AclEntry>(acl))
+
+                val newAttributeView = Files.getFileAttributeView(cachingPath, AclFileAttributeView::class.java)
+                val newOwner = newAttributeView.owner
+                val newAclEntries = newAttributeView.acl
+
+                assertThat(newAttributeView).isInstanceOf(AclFileAttributeView::class.java)
+                // owner doesn't change because we didn't modify the owner in the concurrent file path
+                assertThat(originalOwner).isEqualTo(newOwner)
+                // acl entries do change because we modified the acl entries in the concurrent file path
+                assertThat(originalAclEntries).doesNotContain(newAclEntries)
```

Comment:
This doesn't really seem to prove out `cached acl attributes are modified by concurrent operation using aclFileAttributeView`. This simply proves `cached paths do not cache attribute views`.

I think a better place to document that is in the project README, though the test is fine at demonstrating that.